alpha/veneer/composite/builder.go

```diff
@@ -0,0 +1,287 @@
+package composite
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+const (
+	BasicVeneerBuilderSchema  = "olm.veneer.basic"
+	SemverVeneerBuilderSchema = "olm.veneer.semver"
+	RawVeneerBuilderSchema    = "olm.veneer.raw"
+	CustomVeneerBuilderSchema = "olm.veneer.custom"
+)
+
+type ContainerConfig struct {
+	ContainerTool string
+	BaseImage     string
+	WorkingDir    string
+}
+
+type BuilderConfig struct {
+	ContainerCfg ContainerConfig
+	OutputType   string
+}
+
+type Builder interface {
+	Build(dir string, vd VeneerDefinition) error
+	Validate(dir string) error
+}
+
+type BasicBuilder struct {
+	builderCfg BuilderConfig
+}
+
+var _ Builder = &BasicBuilder{}
+
+func NewBasicBuilder(builderCfg BuilderConfig) *BasicBuilder {
+	return &BasicBuilder{
+		builderCfg: builderCfg,
+	}
+}
+
+func (bb *BasicBuilder) Build(dir string, vd VeneerDefinition) error {
+	// Parse out the basic veneer configuration
+	basicConfig := &BasicVeneerConfig{}
+	err := json.Unmarshal(vd.Config, basicConfig)
+	if err != nil {
+		return fmt.Errorf("unmarshalling basic veneer config: %w", err)
+	}
+	// TODO: should we ensure that the schema matches and error if it does not?
```

Comment:
Yes, please.  We should in general always verify that the input schema matches our expectations w.r.t. type and component fields, as well as erroring on any unexpected fields, like in semver.go:
```
	if err := yaml.Unmarshal(data, &sv, func(decoder *json.Decoder) *json.Decoder {
		decoder.DisallowUnknownFields()
		return decoder
	}); err != nil {
		return nil, err
	}
	if sv.Schema != schema {
		return nil, fmt.Errorf("readFile: input file has unknown schema, should be %q", schema)
	}
```

Reply:
Done in commit https://github.com/operator-framework/operator-registry/pull/1054/commits/606acd5eef65c62ae77a29e0eddaaf55b8332a86

Reply:
Added even better validations for everything in commit https://github.com/operator-framework/operator-registry/pull/1054/commits/f33598f9912c7e2cfa9d55ec79d54c0fd174e1b3